### DISCUSSION TITLE
Rename `$count` to `$chunkSize` in `each()`/`eachById()` for clarity on chunking behavior

### DISCUSSION THREAD

**Discussion by Jehong-Ahn** [laravel/framework#52879] [General]
The `each()` internally uses the `chunk()`. However, the variable name `$count` used as an argument in `each()` can be misleading to users unfamiliar with the internals of Eloquent. Many may incorrectly assume that `$count` refers to the iteration count, when in fact it controls the chunk size.

To improve clarity, I propose renaming the `$count` variable to `$chunkSize`. This change would make it clear that the argument defines the size of the chunk used internally by `chunk()`, reducing potential confusion.

I am attaching a tweet from a user who has also misunderstood the `$count` parameter in the `each()` method.

![image](https://github.com/user-attachments/assets/e8a938ba-e309-4035-8325-1d198f7fcc04)

**Comment by stakx** [comment]
I agree that `chunkSize` would have been the better parameter name, but renaming the parameter now constitutes a breaking change in Eloquent's public API. It's a small one alright, and one that would be easily fixed; but it's still entirely avoidable. I'd therefore recommend not making this otherwise sensible change.

**Reply by shaedrich** [reply]
That's not true:

> [Named arguments](https://www.php.net/manual/en/functions.arguments.php#functions.named-arguments) are not covered by Laravel's backwards compatibility guidelines. We may choose to rename function arguments when necessary in order to improve the Laravel codebase. Therefore, using named arguments when calling Laravel methods should be done cautiously and with the understanding that the parameter names may change in the future.
> [Laravel Docs &#9656; Release Notes § Named arguments](https://laravel.com/docs/13.x/releases#named-arguments)

**Reply by stakx** [reply]
Well, renaming a parameter of a public API still _is_ a breaking change, in the sense that devs using named arguments will have to change their code to keep it functioning. So I as a developer using them would still be pissed at such a change.

That being said, thanks for the quote! It's good to know the Laravel team's position on such changes, they'd be at liberty to go ahead with the suggested rename and I as a named argument-loving dev would only have myself to blame. :-)

**Reply by shaedrich** [reply]
> Well, renaming a parameter of a public API still is a breaking change, in the sense that devs using named arguments will have to change their code to keep it functioning. So I as a developer using them would still be pissed at such a change.

Absolutely right, sorry for misinterpreting your previous reply :-)
